frontend/src/components/App.js

```diff
@@ -16,63 +19,66 @@ class App extends React.Component {
         }
     }
 
+    abortController = new window.AbortController();
+
     componentDidMount() {
-        fetch("http://localhost:5000/api/calendar/", {
-            method: 'POST',
-            body: JSON.stringify(this.state.start)
-        }) //todo move to prop file
+        let url = new URL(ADDRESS + API);
+        let params = [['start', this.state.start.toISOString()]];
+        url.search = new URLSearchParams(params).toString();
+
+        //todo move to prop file
+        fetch(url)
             .then(res => res.json())
             .then(json => {
-                this.setState({events: json})
+                this.setState({events: json});
             });
     }
 
-    async componentDidUpdate(prevProps, prevState) {
-        if (this.state.start !== prevState.start) {
-            let data = {start: this.state.start};
+    componentDidUpdate() {
+        let url = new URL(ADDRESS + API);
+        let params = [['start', this.state.start.toISOString()]];
+        url.search = new URLSearchParams(params).toString();
 
-            //todo move to prop file
-            await fetch("http://192.168.88.254:5000/api/calendar/", {
-                method: 'POST',
-                headers: {
-                    'Content-Type': 'application/json',
-                },
-                body: JSON.stringify(data)
+        fetch(url, {signal: this.abortController.signal})
+            .then(res => res.json())
+            .then(json => {
+                this.setState({events: json});
             })
-                .then(res => res.json())
-                .then(json => {
-                    this.setState({events: json});
-                });
-        }
+            .catch(error => {
+                if (error.name === 'AbortError') return;
+                throw error;
```

Comment:
If you throw an error app will crash. If you want handle errors you can for example setState({ error: true }) and render something when error occurs. Or at least with can set empty list

Reply:
Fixed.